components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/SequenceGenerator.java

```diff
@@ -0,0 +1,377 @@
+                                propertyStr += params[0] + SOAPToRESTConstants.SEQUENCE_GEN.NEW_LINE_CHAR;
+                                JSONObject entry = (JSONObject) ((JSONObject) paramObj).get(paramName);
+                                xPath = (String) entry.get(SOAPToRESTConstants.SEQUENCE_GEN.XPATH);
+                            }
+                            String[] xPathElements = xPath.split(SOAPToRESTConstants.SEQUENCE_GEN.X_PATH_SEPARATOR);
+                            Element prevElement = rootElement;
+                            int elemPos = 0;
+                            for (String xPathElement : xPathElements) {
+                                Element element = doc.createElementNS(namespace,
+                                        SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_PREFIX
+                                                + SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_SEPARATOR + xPathElement);
+                                if (doc.getElementsByTagName(element.getTagName()).getLength() > 0) {
+                                    prevElement = (Element) doc.getElementsByTagName(element.getTagName()).item(0);
+                                } else {
+                                    if (elemPos == xPathElements.length - 1) {
+                                        element.setTextContent(
+                                                SOAPToRESTConstants.SEQUENCE_GEN.PROPERTY_ACCESSOR + count);
+                                        count++;
+                                    }
+                                    prevElement.appendChild(element);
+                                    prevElement = element;
+                                }
+                                elemPos++;
+                            }
+                        }
+                    } else if (jsonObject.get(obj) instanceof JSONObject) {
+                        JSONObject param = (JSONObject) jsonObject.get(obj);
+                        Element element = doc.createElementNS(namespace, SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_PREFIX
+                                        + SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_SEPARATOR + param.get(SOAPToRESTConstants.SWAGGER.NAME).toString());
+                        element.setTextContent(SOAPToRESTConstants.SEQUENCE_GEN.PROPERTY_ACCESSOR + count);
+                        rootElement.appendChild(element);
+                        String paramElements = createParameterElements(param.get(SOAPToRESTConstants.SWAGGER.NAME).toString(),
+                                SOAPToRESTConstants.PARAM_TYPES.QUERY);
+                        String[] params = paramElements.split(SOAPToRESTConstants.SEQUENCE_GEN.COMMA);
+                        argStr += params[1] + SOAPToRESTConstants.SEQUENCE_GEN.NEW_LINE_CHAR;
+                        propertyStr += params[0] + SOAPToRESTConstants.SEQUENCE_GEN.NEW_LINE_CHAR;
+                        count++;
+                    }
```

Comment:
Shall we include debug logs in-between bigger methods. This will be helpful in-terms of troubleshooting later

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2